Gems/Atom/RPI/Code/Source/RPI.Builders/Pass/PassBuilder.cpp

```diff
@@ -65,35 +65,45 @@ namespace AZ
             m_isShuttingDown = true;
         }
 
-        void PassBuilder::CreateJobs(const AssetBuilderSDK::CreateJobsRequest& request, AssetBuilderSDK::CreateJobsResponse& response) const
+        // --- Code related to dependency shader asset handling ---
+
+        // Helper class to pass parameters to the AddDependency and FindReferencedAssets functions below
+        struct FindPassReferenceAssetParams
         {
-            if (m_isShuttingDown)
-            {
-                response.m_result = AssetBuilderSDK::CreateJobsResultCode::ShuttingDown;
-                return;
-            }
+            void* passAssetObject;
+            Uuid passAssetUuid;
+            SerializeContext* serializeContext;
+            AZStd::string_view passAssetSourceFile;     // File path of the pass asset
+            AZStd::string_view dependencySourceFile;    // File pass of the asset the pass asset depends on
+            const char* jobKey;                         // Job key for adding job dependency
```

Comment:
it's recommended in coding standards to initialize variables in the class/struct declaration. The pointers in this struct should all be initialized to nullptr.